modules/unpickler/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/Scala3Unpickler.scala

```diff
@@ -57,17 +59,48 @@ class Scala3Unpickler(
       case None =>
         throw new Exception(s"Cannot find Scala symbol of $fqcn")
       case Some(declaringType) =>
-        val matchingSymbols =
-          declaringType.declarations
-            .collect { case sym: TermSymbol if sym.isTerm => sym }
-            .filter(matchSymbol(method, _))
-
-        if matchingSymbols.size > 1 then
-          val message =
-            s"Found ${matchingSymbols.size} matching symbols for $method:" +
-              matchingSymbols.mkString("\n")
-          throw new Exception(message)
-        else matchingSymbols.headOption
+        matchesLocalMethodOrLazyVal(method) match
+          case Some((methodName, index)) => findLocalMethodOrLazyVal(declaringType, methodName, index)
+          case _ =>
+            val matchingSymbols = declaringType.declarations
+              .collect { case sym: TermSymbol if sym.isTerm => sym }
+              .filter(matchSymbol(method, _))
+            if matchingSymbols.size > 1 then
+              val message = s"Found ${matchingSymbols.size} matching symbols for $method:" +
+                matchingSymbols.mkString("\n")
+              throw new Exception(message)
+            else matchingSymbols.headOption
+
+  def findLocalMethodOrLazyVal(declaringType: DeclaringSymbol, name: String, index: Int): Option[TermSymbol] =
+    var currentindex = 1
+    val declaringtpe = declaringType.owner match
+      case s: DeclaringSymbol => s
+      case _ => declaringType
+
+    declaringtpe.declarations
+      .flatMap(sym =>
+        sym.tree match
+          case Some(tree) =>
+            tree.walkTree(tree =>
+              if currentindex <= index then
+                tree match
+                  case DefDef(_, _, _, _, symbol) =>
+                    if matchTargetName(name, symbol) && currentindex == index then List(symbol)
+                    else
+                      if matchTargetName(name, symbol) then currentindex += 1
+                      List()
+                  case ValDef(_, _, _, symbol) =>
+                    if matchTargetName(name, symbol) && currentindex == index then List(symbol)
+                    else
+                      if matchTargetName(name, symbol) then currentindex += 1
+                      List()
+
+                  case _ => List()
+              else List()
+            )((l1, l2) => l1 ++ l2, List())
+          case None => List()
+      )
+      .headOption
```

Comment:
The following would be slightly less performant but easier to read.

```suggestion
    val matchingSymbols = declaringtpe.declarations
      .flatMap(sym => sym.tree)
      .flatMap(tree =>
        tree.walkTree {
          case DefDef(_, _, _, _, symbol) if matchTargetName(name, symbol) => List(symbol)
          case ValDef(_, _, _, symbol) if matchTargetName(name, symbol) => List(symbol)
          case _ => List.empty
        }(_ ++ _, List.empty)
      )
    Some(matchingSymbols(index - 1))
```